V2 rev18

votan73 opened the pull request:
All changes for Rev.18.

votan73 committed:
Rev. 18

votan73 committed:
rev. 18:
- Did TODO: Removed API10025 code.
- Additional hooked SetMapToAutoMapNavigationTargetPosition.
- Save measurements to Saved Var: LibGPS_Data
- Made LibStub optional.
- Ignore return code of CalculateMapMeasurements. Return SET_MAP_RESULT_MAP_CHANGED to refresh map, no matter there it is now.

sirinsidiator commented on src/LibGPS.lua:
```diff
@@ -1,790 +1,791 @@
+------------------------------------------------------------------
+
+local LIB_NAME = "LibGPS2"
+local lib = LibStub and LibStub:NewLibrary(LIB_NAME, 18)
```
Please keep the old code for the build script.

sirinsidiator commented on src/LibGPS.lua:
```diff
@@ -1,790 +1,791 @@
+
+local function InitializeSaveData()
+    local VERSION = 2
+    LibGPS_Data = LibGPS_Data or {APIVersion = GetAPIVersion(), Version = VERSION}
```
I'm using the same saved variable with different fields in the v3 branch:
```
        saveData = {
            version = GetAPIVersion(),
            measurements = {}
        }
```
Having a separate version for the format and api looks good and I'll add that too. As for the field names, I'd prefer them to start with lower case letters (apiVersion and version).

sirinsidiator commented on src/LibGPS.txt:
```diff
@@ -1,15 +1,16 @@
-LibGPS.lua
+## Title: LibGPS
+## Author: sirinsidiator & votan
+## Version: 2.0 r18
```
please keep the token for the build script

sirinsidiator commented on src/LibGPS.txt:
```diff
@@ -1,15 +1,16 @@
+## Author: sirinsidiator & votan
+## Version: 2.0 r18
+## APIVersion: 100029 100030
+## AddOnVersion: 18
```
same as above

sirinsidiator commented on src/LibMapPing/LibMapPing.lua:
```diff
@@ -1,450 +1,451 @@
-Load()
-
-LibMapPing = lib
+local LIB_IDENTIFIER = "LibMapPing"
```
Should we just get rid of the bundled LMP?

votan73 committed:
- keep build script markers.

votan73 committed:
rev. 18:
- Removed bundled LibMapPing
- Depend on LibMapPing 900+
- Lowercase member names.

votan73 committed:
rev. 18:
- keep build script markers.

votan73 committed:
removed BOM
